src/controllers/register.ts

```diff
@@ -113,16 +115,35 @@ export const post = async function (req: Request, res: Response) {
 
         let redirectTo = "/register-confirm";
 
-        const sheetsService: SheetsService = new SheetsService(process.env.REGISTER_SPREADSHEET_ID as string);
-        await sheetsService.init().catch(() => (redirectTo = "/register-error"));
-        await sheetsService
-            .appendValues(values, process.env.REGISTER_SHEET_DATA_RANGE as string, process.env.REGISTER_SHEET_HEADER_RANGE as string)
-            .catch(reason => {
-                console.log(reason);
-                redirectTo = "/register-error";
-            });
+        if (process.env.GOOGLE_SHEETS_INTEGRATION_ENABLED !== "false") {
+            const sheetsService: SheetsService = new SheetsService(process.env.REGISTER_SPREADSHEET_ID as string);
+            await sheetsService.init().catch(() => (redirectTo = "/register-error"));
+            await sheetsService
+                .appendValues(values, process.env.REGISTER_SHEET_DATA_RANGE as string, process.env.REGISTER_SHEET_HEADER_RANGE as string)
+                .catch(reason => {
+                    console.log(reason);
+                    redirectTo = "/register-error";
+                });
+            console.log("Saved to sheets");
+        }
 
-        console.log("Saved to sheets");
+        if (process.env.JIRA_INTEGRATION_ENABLED === "true") {
+            console.log("Using Jira integration");
+
+            try {
+                const jiraService = new JiraTicketService(values);
+                await jiraService.sendJiraTicket();
+            } catch (error) {
+                if (axios.isAxiosError(error)) {
+                    console.error("Jira Post request failed with status code: ", error.status);
+                    console.error("Axios Error, errorMessages Object: ", error.response?.data?.errorMessages);
+                    console.error("Axios Error, errors Object: ", error.response?.data?.errors);
+                } else {
+                    console.error("Jira Integration error: ", error);
```

Comment:
When logging errors please make sure we are not logging PII data.

Reply:
Please squash commits

Reply:
@Ryan-Andrews99 you happy to address this one?

Reply:
I think the error logging here strikes the right balance. We're only logging parts of the Axios error which will not log the POST payload which all the PII. The `errors` and `errorMessages` objects _may_ contain PII, depending on the Jira API response but they also will likely contain very useful information about why a request might've been rejected. Given we've caught the Axios error in the `if` the only other errors we're likely to log are missing environment variable names, which do not contain PII.